src/module_3/module_3.py

```diff
@@ -0,0 +1,129 @@
+# %% Import packages
+import pandas as pd
+
+from sklearn.model_selection import train_test_split
+from sklearn import linear_model
+from sklearn.linear_model import Ridge
+from sklearn.ensemble import GradientBoostingClassifier
+from sklearn.ensemble import RandomForestClassifier
+from sklearn import metrics
+from sklearn.preprocessing import normalize
+
+import matplotlib.pyplot as plt
+
+# %% Load data
+address = "D:\\Yo\\"
+filename = "feature_frame.csv"
+df = pd.read_csv(address + filename)
+
+# %% Data preprocessing
+
+# Obtain the id's for the orders where 5 or more products where bought
+filter_df = df[df["outcome"]==1]
+filter_df = filter_df.groupby("order_id").filter(lambda x: len(x) > 4)
+
+# Filter the data so we retain only those ids
+dataset = df[df["order_id"].isin(filter_df["order_id"])]
+
+# Change date format to retain only year, month and day
+dataset["created_at"] = dataset["created_at"].str[:-9]
+dataset["created_at"] = dataset["created_at"].str.replace("-","")
+pd.to_numeric(dataset["created_at"])
+dataset["order_date"] = dataset["order_date"].str[:-9]
+dataset["order_date"] = dataset["order_date"].str.replace("-","")
+pd.to_numeric(dataset["order_date"])
+
+# Order by date
+dataset = dataset.sort_values(by=["order_date"])
+
+# %% Drop qualitative features
+print(dataset.columns)
+print(dataset.dtypes)
+dataset.drop(["product_type" , "vendor"], axis=1, inplace=True)
```

Comment:
I prefer to put it as `dataset.drop(columns = ["product_type" , "vendor"], inplace=True)` in order to be more explicit